### PR TITLE
Add automatic module name

### PR DESCRIPTION
**Pull request by afloarea**
Signed-off-by: afloarea <adrianfloarea07@yahoo.com>

Motivation:

Add automatic module name as per issue [#524](https://github.com/vert-x3/issues/issues/524)

### CODE REVIEW

**Comment by vietj** [comment]
I woiuld like first to see if we can do something in the vertx-parent pom

**Comment by vietj** [comment]
@afloarea please have a look at https://github.com/vert-x3/vertx-parent/commit/1384372552de56622b7455d79ce610b5a3b3c1a9

This change should allow to have a file named `MANIFEST.MF` that will contain the entry containing `Automatic-Module-Name: io.vertx.core` that will define the automatic module name.

File: src/main/resources/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Automatic-Module-Name: io.vertx.core
+
```
Comment (afloarea) [inline src/main/resources/META-INF/MANIFEST.MF:2]: Note: The new line is required. Otherwise the maven plugin does not work for some reason.

Comment (vietj) [reply src/main/resources/META-INF/MANIFEST.MF:2]: shall we upgrade the plugin version ?

Comment (vietj) [reply src/main/resources/META-INF/MANIFEST.MF:2]: I mean would it solve the issue ?

Comment (afloarea) [reply src/main/resources/META-INF/MANIFEST.MF:2]: I tried with the newest I could find on maven central (maven jar plugin 3.2.0) and the issue is still present

Comment (afloarea) [reply src/main/resources/META-INF/MANIFEST.MF:2]: There is no error or anything, the jar is built but the Automatic-Module-Name property is not in the built jar.

File: pom.xml [hunk 2]
```diff
@@ -56,6 +56,7 @@
     <jmh.version>1.19</jmh.version>
     <vertx.testNativeTransport>false</vertx.testNativeTransport>
     <vertx.testDomainSockets>false</vertx.testDomainSockets>
+    <jar.manifest>src/main/resources/META-INF/MANIFEST.MF</jar.manifest>
```
Comment (vietj) [inline pom.xml:59]: use rather `${project.basedir}/src/main/resources/META-INF/MANIFEST.MF`

File: src/main/resources/META-INF/MANIFEST.MF (hunk 1 above)
Comment (vietj) [reply src/main/resources/META-INF/MANIFEST.MF:2]: ok then

File: pom.xml (hunk 2 above)
Comment (afloarea) [reply pom.xml]: done

**Pull request merged**